arangod/RocksDBEngine/RocksDBDumpManager.cpp

```diff
@@ -0,0 +1,175 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#include "RocksDBDumpManager.h"
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/Exceptions.h"
+#include "Basics/system-functions.h"
+#include "Cluster/ServerState.h"
+#include "RestServer/DatabaseFeature.h"
+#include "RocksDBEngine/RocksDBEngine.h"
+#include "RocksDBEngine/RocksDBDumpContext.h"
+#include "VocBase/ticks.h"
+
+#include <absl/strings/str_cat.h>
+
+using namespace arangodb;
+
+RocksDBDumpManager::RocksDBDumpManager(RocksDBEngine& engine)
+    : _engine(engine) {}
+
+RocksDBDumpManager::~RocksDBDumpManager() {
+  std::lock_guard mutexLocker{_lock};
+  _contexts.clear();
+}
```

Comment:
Using locks in the destructor is a code smell. In almost all cases they are either unnecessary or they actually don't protect you from a race after all. In this case the lock is unnecessary.

Reply:
Yes, I agree!